experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -126,21 +122,59 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 				Stages: map[string]*compositionv1alpha1.StageStatus{},
 			},
 		}
-		if err := ctrl.SetControllerReference(&inputcr, &plancr, r.Scheme); err != nil {
+		if err := ctrl.SetControllerReference(inputcr, &plancr, r.Scheme); err != nil {
 			logger.Error(err, "Unable to set controller reference for Plan Object")
```

Comment:
Lets log the error.

Reply:
logger.Error is passed the err object. Iam assuming that will be logged. Added the plan name and namespace.